task/net/README.md

```diff
@@ -1,6 +1,155 @@
 # About
 The `net` task implements a small netstack based on [_smoltcp_](https://github.com/smoltcp-rs/smoltcp)
 
+# Design of the SP Network Stack
+
+NOTE: This network stack is specialized to the SP's needs and is not, at the
+moment, intended as a fully general network stack.
+
+In particular, we currently only support IP and UDP, with a focus on IPv6.
+
+## The Net Task
+
+The `net` task is responsible for maintaining system network state. It includes:
+
+1. The Ethernet MAC and PHY drivers.
+2. Ethernet DMA support and buffer management.
+3. The IP stack itself (`smoltcp`).
+4. Buffers for each socket defined in the system.
+
+The `net` task itself is designed to manage network interface events as quickly
+as possible, without blocking on other tasks. It sends no IPCs during its normal
+operation.
```

Comment:
```suggestion
operation, except when necessary to configure board-specific hardware (e.g.
external switches or PHYs).
```

Reply:
Hm. I had forgotten that it grew that behavior. Will update.